src/urbit/lib/chess.hoon

```diff
@@ -865,6 +941,28 @@
   ++  draw-claimable  ::  XX no threefold repetition yet
     ^-  ?
     (legal-move [%end %'½–½'])
+  ++  check-repetition
+    |=  [fen-repetition=(map @t @ud) fen=@t]
+    ^-  (map @t @ud)
+    ?.  (~(has by fen-repetition) fen)
+      %=  $
+        fen-repetition  (~(put by fen-repetition) fen 1)
+      ==
+    ?.  (gte 2 (~(got by fen-repetition) fen))
+      %=  $
+        fen-repetition  (~(put by fen-repetition) fen 2)
+      ==
+      %=  $
+        fen-repetition  (~(put by fen-repetition) fen +((~(got by fen-repetition) fen)))
+      ==
+  ++  check-threefold
+    |=  [fen-repetition=(map @t @ud) fen=@t]
+    ^-  ?
+    ?.  (~(has by fen-repetition) fen)
+      |
+    ?.  (gte 2 (~(got by fen-repetition) fen))
+      &
+      |  
```

Comment:
There's a simpler way to write this, and also a bug (`gte` instead of `gth`):
```
|=  [fen-repetition=(map @t @ud) fen=@t]
  ^-  ?
  =*  count  (~(get by fen-repetition) fen)
  ?~  count
    |
  (gth 2 (need count))
```